core/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java

```diff
@@ -251,6 +250,28 @@ public static ElasticsearchParseException newConfigurationException(String proce
         return processors;
     }
 
+    public static TemplateService.Template compileTemplate(String processorType, String processorTag, String propertyName,
+                                                           String propertyValue, TemplateService templateService) {
+        try {
+            return templateService.compile(propertyValue);
+        } catch (Exception e) {
```

Comment:
@jasontedor Would using `ExceptionsHelper#convertToElastic(...)` helper method in `ConfigurationUtils#newConfigurationException(...)` or similar here be sufficient?

I think the excepting being thrown if the mustache template couldn't be compiled is a `MustacheException`, which gets wrapped by `GeneralScriptException` in `ScriptService#compileInternal(...)`, which is what actually should be fixed. Instead the lang-mustache should catch `MustacheException` exceptions with `ScriptException`.